Alert for SHF Application missing uploaded files (no files uploaded)

weedySeaDragon opened the pull request:
### PT Story: email to applicant: if 0 files were uploaded, remind them
https://www.pivotaltracker.com/story/show/153947622

### Changes proposed in this pull request:
1.  This will _not_ check applications that are `accepted` or `rejected.`  (This is the correct action.)
2.  Updated the locale words/entries per Suss's input and review.

2. Created ~DRAFT locale words (placeholders)~
3. ~DRAFT email~
4. DRAFT timing to send them out
5. Added to load_conditions

### Screenshots (Optional):
<img width="679" alt="email-applicant-no-files-login-link2" src="https://user-images.githubusercontent.com/673794/52434115-76b3e080-2ac3-11e9-83a5-cb19486c3923.png">


### Ready for review:
@thesuss 
@patmbolger

weedySeaDragon committed:
DRAFT locale wording for "no uploaded files" reminder email

weedySeaDragon committed:
DRAFT locale wording for "no uploaded files" reminder email

weedySeaDragon committed:
DRAFT mail for no files uploaded

weedySeaDragon committed:
DRAFT mail for no files uploaded

weedySeaDragon committed:
No files uploaded Alert

weedySeaDragon committed:
No files uploaded Alert

weedySeaDragon committed:
feature for testing 'no files uploaded' alert

weedySeaDragon committed:
feature for testing 'no files uploaded' alert

weedySeaDragon committed:
use global for the states to check; ensure we're checking Strings in tests

weedySeaDragon committed:
use global for the states to check; ensure we're checking Strings in tests

weedySeaDragon committed:
add to load_conditions

weedySeaDragon committed:
add to load_conditions

thesuss commented:
First question: That is correct

thesuss commented on config/locales/en.yml:
```diff
@@ -1015,6 +1015,12 @@ en:
     member_mailer:
+      app_no_uploaded_files:
+        subject: Please upload qualification files to your Sveriges Hundföretagare application
+        message_text: "This is a reminder that you need up load files for your SHF application....  why they're needed..."
```
This is a reminder that you need up load files for your SHF application so that it can be reviewed.

thesuss commented on config/locales/sv.yml:
```diff
@@ -1018,6 +1018,12 @@ sv:
 
     member_mailer:
+      app_no_uploaded_files:
+        subject: Please upload qualification files to your Sveriges Hundföretagare application
```
Din ansökan till Sveriges Hundföretagare saknar underlag

thesuss commented on config/locales/sv.yml:
```diff
@@ -1018,6 +1018,12 @@ sv:
     member_mailer:
+      app_no_uploaded_files:
+        subject: Please upload qualification files to your Sveriges Hundföretagare application
+        message_text: "This is a reminder that you need up load files for your SHF application....  why they're needed..."
```
Detta är en automatisk påminnelse från systemet som meddelar att din ansökan saknar uppladdade filer. Filerna behövs för att vi ska kunna handlägga din ansökan.

thesuss commented on config/locales/sv.yml:
```diff
@@ -1018,6 +1018,12 @@ sv:
+      app_no_uploaded_files:
+        subject: Please upload qualification files to your Sveriges Hundföretagare application
+        message_text: "This is a reminder that you need up load files for your SHF application....  why they're needed..."
+        thanks:  "Thank you for wanting to be a member and contributing to a Sweden with an ethical dog industry."
```
Tack för att du vill vara medlem.

weedySeaDragon committed:
Merge branch 'develop' of https://github.com/AgileVentures/shf-project into sprint60-shfapp-missing-uploads-#153947622

weedySeaDragon commented:
@thesuss -  What about including this in the email?

När du har loggat in visas en knapp som tar dig till din ansökan, och sedan kan du ladda upp dina filer.

(English:  "After you log in, you will see a button that will take you to your application, and then you can upload your files.")

I updated the email screenshots in the PR description above.  There is 1 version _without_ this sentence and another version _with_ the sentence.

weedySeaDragon committed:
remove FactoryBot creation from the preview method. (it caused too much weirdness)

weedySeaDragon committed:
remove FactoryBot creation from the preview method. (it caused too much weirdness)

weedySeaDragon committed:
locale entries per Susanna

weedySeaDragon committed:
locale entries per Susanna

weedySeaDragon committed:
add draft entry for .. after you log in, you will see a button ... to your app....

weedySeaDragon committed:
add draft entry for .. after you log in, you will see a button ... to your app....

weedySeaDragon committed:
email body _without_ the "after you log in you will see a button to your app" (for now)

weedySeaDragon committed:
email body _without_ the "after you log in you will see a button to your app" (for now)

weedySeaDragon committed:
Merge branch 'av-develop' into sprint60-shfapp-missing-uploads-#153947622

# Conflicts:
#	lib/tasks/conditions/load_conditions.rake

weedySeaDragon committed:
standardize link to login page (for emails)

weedySeaDragon committed:
standardize link to login page (for emails)

weedySeaDragon committed:
use milliseconds to create unique email

preview will actually call each method twice: once for the header and once for the body. If it happens fast enough, it will happen in less than 1 second, and so the 2 email strings created ("new_email" done twice) won't be unique.  Thus we have to use milliseconds to create a unique email name.

weedySeaDragon committed:
use milliseconds to create unique email

preview will actually call each method twice: once for the header and once for the body. If it happens fast enough, it will happen in less than 1 second, and so the 2 email strings created ("new_email" done twice) won't be unique.  Thus we have to use milliseconds to create a unique email name.

weedySeaDragon committed:
add link to where they can login and instructions

weedySeaDragon committed:
add link to where they can login and instructions

weedySeaDragon committed:
remove "SHF" from the 'click here to login...' text

weedySeaDragon committed:
remove "SHF" from the 'click here to login...' text

weedySeaDragon commented:
Will integrate this with #615 when that's done

weedySeaDragon committed:
Alert for membership lapsed  (#605)

* remove old block comments

* DRAFT locale entries for Membership lapsed email

* email for Membership lapsed alert; mailer method

* requirements for Is a membership lapsed? or not

* delete old placeholder for expired (lapsed) membership

* deleted old placeholder Membership expired spec

* MembershipLapsedAlert spec and feature

* Load Condition for Membership Lapsed alert

* create member with membership lapsed for mail preview

* don't use FactoryBot to create users to preview. Causes too many problems

* update Locale text entries per Suss

* add User scope .current_members

* add a factory for users  :member_with_expiration_date

* add style for highlighting dates that are important (past due, upcoming, etc)

* clarify comment

* create standardize message and partial for link to member's account page

* use common partial for showing link to member's account page

* style changes per Hound suggestions

* fix extra spaces in HAML comment

patmbolger committed:
User selects how application files will be delivered (#615)

* final - refactored

* merged upstream/develop

* interim progress

* most non-test functionality built

* some tests complete, footnotes for file delivery selection

* add name to FDM model, FDM helper returns footnotes

* corrected test description

* unit tests working

* progress on tests

* more test changes

* refined missing-files logic in controller

* refactor FDM model and App factory

* added FDM scope test

* clean up tests

* progress on cuke tests

* added cuke tests

* correct save/submit button enable prompt

* progress in converting cuke tests

* more progress converting cuke tests

* refactor controller for better UX

* converting cike tests; fix JS bug

* marked some cuke tests as ci-skip-test

* remove unneeded rake task

* hound issues

* optimize JS for shf-app

* changes from review comments

* changes due to review

* more changes from review

* changed ** to !! in sv.yml to avoid regexp problems

* fixed unit test

weedySeaDragon committed:
conditional logic based on delivery_method (spec = draft)

weedySeaDragon committed:
preview: draft to show app with no uploaded files; needs variations

patmbolger reviewed (approved):
Looks good!

weedySeaDragon committed:
Alert: company info incomplete  (#616)

* reorganize: class methods at top per RuboCop standards; add whitespace

* DRAFT locale entries

* parent class for Requirements that are just the opposite of another requirement

* refactor so it's a subclass of the AbstractOppositeRequirements class

* add step that sets the company name to a blank

* add Requirements for Company Info Complete (or not complete) classes

* add method complete? (uses Requirements class); branding_start_date

* update h-branding-fee-not-due spec to check for opposite class

* specs for RequirementsForCoInfo[Not]Complete classes

* add email view and mailer method + spec and preview

* remove branding_start_date; no needed for Alert

* feature for Company info incomplete alert

* Company info not complete alert and spec

* add company_info_incomplete to Mailer preview

* correct the comment for CompanyInfoIncompleteAlert

* add to load_conditions; make schedule arrays std

* locale entry updates per Susanna

* add method missing_region?

* use Company method missing_region? so implementation is hidden

* partial for 'you may need to login first' with link to login page

* shared spec for 'you may need to login first'; will be used in mult. emails

* list what is missing in the email; spec, draft locale entries

* update preview with example missing name and region

* sv locale entry for company_needs_name -> foretag

* locale updates: co_name_missing, co_region_missing

* comments: if def. of 'complete' changes, changes in 2 places are required

* fix comments in AbstractOppositeRequirements class

weedySeaDragon committed:
create the membership sequence if needed (e.g. after DatabaseCleaner start)

fix for tests that sometime fail because the DatabaseCleaner has removed the sequence and it is not automatically recreated.

weedySeaDragon committed:
complete the spec for the alert

weedySeaDragon committed:
Merge branch 'sprint60-shfapp-missing-uploads-#153947622' of https://github.com/weedySeaDragon/shf-project into sprint60-shfapp-missing-uploads-#153947622

# Conflicts:
#	app/mailers/member_mailer.rb
#	app/models/conditions_response/shf_app_no_uploaded_files_alert.rb
#	config/locales/en.yml
#	config/locales/sv.yml
#	spec/mailers/member_mailer_spec.rb
#	spec/mailers/previews/member_mailer_preview.rb
#	spec/models/conditions_response/shf_app_no_uploaded_files_alert_spec.rb
